### ISSUE TITLE
Problem in subscription after cryptoexchange.net-beta merge

### ISSUE THREAD

**Issue by christose** [JKorf/Binance.Net#902]
The Cryptoexchange.net-beta merge which contains the renaming of all functions to "Async" caused an issue with subscriptions, .Net replies back with "The semaphore has been disposed" the error obviously comes after calling => BinanceSocketClientSpot.cs => return await Subscribe(symbols, handler).ConfigureAwait(false);

**Comment by JKorf** [comment]
I can't reproduce this, could you share some code which triggers the error?

**Comment by christose** [comment]
Yes here's the code that triggers the error, the first attempt to subscribe
```
foreach (var symbol in symbols)
{
   Task.Run(async () =>
   {
        try
        {
             var subscription = await BinanceSocketClient.Spot.SubscribeToTradeUpdatesAsync(symbol.Instrument, OnTradeUpdateMessage);
             if (!subscription.Success)
             ......

```
This issue appeared since the merge on the 12th of August.
I have also tried to isolate all other parallel subscriptions and calls to the API and execute the above piece of code and I get the same error and with the latest master branch from few days ago.

**Comment by christose** [comment]
Hey @JKorf problem solved, after I have cleaned the solution and all the projects, Binance.Net included; and re-builded everything, that issue went away. Sorry for bothering you for nothing.
